storage/block/block.go

```diff
@@ -238,10 +344,10 @@ func (b *dbBlock) resetSegmentWithLock(seg ts.Segment) {
 	b.length = seg.Len()
 	b.checksum = digest.SegmentChecksum(seg)
 
-	b.retriever = nil
```

Comment:
The general idea now is that you should keep your retriever around so that if you want to be unloaded you can as you still have a reference to it.